library/ssl_tls13_generic.c

```diff
@@ -1402,28 +1402,47 @@ int mbedtls_ssl_tls13_write_change_cipher_spec(mbedtls_ssl_context *ssl)
  *
  * struct {
  *   select ( Handshake.msg_type ) {
- *     ...
+ *     case new_session_ticket:   uint32 max_early_data_size;
  *     case client_hello:         Empty;
  *     case encrypted_extensions: Empty;
  *   };
  * } EarlyDataIndication;
+ *
+ * We use `mbedtls_ssl_is_handshake_over()` to decide if `max_early_data_size`
+ * should be sent for `new_session_ticket` is post-handshake message.
  */
 #if defined(MBEDTLS_SSL_EARLY_DATA)
 int mbedtls_ssl_tls13_write_early_data_ext(mbedtls_ssl_context *ssl,
                                            unsigned char *buf,
                                            const unsigned char *end,
-                                           size_t *out_len)
+                                           size_t *out_len,
+                                           const mbedtls_ssl_session *session)
```

Comment:
`session->max_early_data_size` is set in `ssl_write_new_session_ticket_body()` before `f_write_ticket()` called. 

Add this parameter is to make sure `max_early_data_size` of client/server are same